test/core/test/jest-expect.test.ts

```diff
@@ -903,24 +903,152 @@ it('correctly prints diff with asymmetric matchers', () => {
   }
 })
 
-it('toHaveProperty error diff', () => {
-  setupColors(getDefaultColors())
+// make it easy for dev who trims trailing whitespace on IDE
+function trim(s: string): string {
+  return s.replaceAll(/ *$/gm, '')
+}
 
-  // make it easy for dev who trims trailing whitespace on IDE
-  function trim(s: string): string {
-    return s.replaceAll(/ *$/gm, '')
+function getError(f: () => unknown) {
+  try {
+    f()
+    return expect.unreachable()
   }
-
-  function getError(f: () => unknown) {
-    try {
-      f()
-      return expect.unreachable()
-    }
-    catch (error) {
-      const processed = processError(error)
-      return [processed.message, trim(processed.diff)]
-    }
+  catch (error) {
+    const processed = processError(error)
+    return [processed.message, trim(processed.diff)]
   }
+}
+
+it('toMatchObject error diff', () => {
+  setupColors(getDefaultColors())
+
+  // single property on root (3 total properties, 1 expected)
+  expect(getError(() => expect({ a: 1, b: 2, c: 3 }).toMatchObject({ c: 4 }))).toMatchInlineSnapshot(`
+    [
+      "expected { a: 1, b: 2, c: 3 } to match object { c: 4 }
+    (2 more properties properties in actual)",
```

Comment:
It says "properties properties" 🤔 

Also not sure about a few things:

- Shouldn't it be under the object?
- The text itself is not very descriptive - can we put more information here? Like "2 more properties not shown since they are the same" (or maybe there is a better way to tell it more concisely?)

Reply:
Thanks, missed the duplicate "properties" when I introduced the `pluralize` function. 

Changed the message to:

```
1 matching property omitted from actual
5 matching properties omitted from actual
...etc.
```

I think that expresses it more clearly that x matching properties from the actual were removed.

As for placing it under the object, that's a little bit more convoluted. The error is caught and processed (`processError` in `error.ts`) and then rendered later on. Is there a way to influence that behavior to render it beneath?